### ISSUE TITLE
Multiple `cloudflare_list_item` cannot be a substring of an existing entry on the list

### ISSUE THREAD

**Issue by patrick-vares-stash** [cloudflare/terraform-provider-cloudflare#3506]
### Confirmation

- [X] This is a bug with an existing resource and is not a feature request or enhancement. Feature requests should be submitted with Cloudflare Support or your account team.
- [X] I have searched the issue tracker and my issue isn't already found.
- [X] I have replicated my issue using the latest version of the provider and it is still present.

### Terraform and Cloudflare provider version

```
Terraform v1.9.2
on darwin_arm64
+ provider registry.terraform.io/cloudflare/cloudflare v4.38.0
```

### Affected resource(s)

`cloudflare_list_item`

### Terraform configuration files

```hcl
resource "cloudflare_list" "test_bug" {
  account_id  = var.account_id
  name        = "bug_list"
  description = "list for testing a bug"
  kind        = "hostname"
}

resource "cloudflare_list_item" "item_1" {
  account_id = var.account_id
  list_id    = cloudflare_list.test_bug.id
  comment    = "item substring"
  hostname {
    url_hostname = "bug.test"
  }
}


resource "cloudflare_list_item" "item_2" {
  account_id = var.account_id
  list_id    = cloudflare_list.test_bug.id
  comment    = "item superstring"
  hostname {
    url_hostname = "differing-bug.test"
  }
}
```

### Link to debug output

https://gist.github.com/patrick-vares-stash/5910e2100e4ded69566145bffbe522db

### Expected output

Successful create of 3 resources:
- `cloudflare_list.test_bug`
- `cloudflare_list_item.item_1`
- `cloudflare_list_item.item_2`

### Actual output

Successful create of 2 resources:
- `cloudflare_list.test_bug`
- `cloudflare_list_item.item_2`

Failed create of 1 resource:
- `cloudflare_list_item.item_1`

```
2024-07-25T14:50:11.964-0400 [ERROR] vertex "cloudflare_list_item.item_1" error: failed to create list item: <unknown>

Error: failed to create list item: <unknown>

  with cloudflare_list_item.item_1,
  on test_list.tf line 8, in resource "cloudflare_list_item" "item_1":
   8: resource "cloudflare_list_item" "item_1" {

failed to match exactly one list item
```

### Steps to reproduce

1. Create a `cloudflare_list` with `kind = "hostname"`
2. Create a `cloudflare_list_item` with `hostname.url_hostname = "some.string"`
3. Create a `cloudflare_list_item` with `hostname.url_hostname = "longer-some.string"`
4. Run `terraform apply` and observe that Terraform is unable to successfully complete the `apply`
5. Note that all `cloudflare_list_item`s are actually present within the CloudFlare console

### Additional factoids

A screenshot of the created list within the CloudFlare console after applying the failing terraform above

![image](https://github.com/user-attachments/assets/ade1852f-396b-4554-80f2-066cfd99fa9a)

**Comment by jacobbednarz** [comment]
we've patched this for other types in #3368 and #3335. if you want to send over a PR for the fix here, i'll take a look.

**Comment by patrick-vares-stash** [comment]
Sure: https://github.com/cloudflare/terraform-provider-cloudflare/pull/3515
